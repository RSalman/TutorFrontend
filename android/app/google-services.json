{
  "project_info": {
    "project_number": "779865862095",
    "firebase_url": "https://prospr-8a50d.firebaseio.com",
    "project_id": "prospr-8a50d",
    "storage_bucket": "prospr-8a50d.appspot.com"
  },
  "client": [
    {
      "client_info": {
        "mobilesdk_app_id": "1:779865862095:android:1b83d4b60becf7dd",
        "android_client_info": {
          "package_name": "com.tutorfrontend"
        }
      },
      "oauth_client": [
        {
          "client_id": "779865862095-4tk4rhhj654t9tepp8c6i5elqlaj6sgg.apps.googleusercontent.com",
          "client_type": 3
        }
      ],
      "api_key": [
        {
          "current_key": "AIzaSyBDjtDNEYE5WbvPy3VumSoq1dEDhKFcOH8"
        }
      ],
      "services": {
        "analytics_service": {
          "status": 1
        },
        "appinvite_service": {
          "status": 1,
          "other_platform_oauth_client": []
        },
        "ads_service": {
          "status": 2
        }
      }
    }
  ],
  "configuration_version": "1"
}