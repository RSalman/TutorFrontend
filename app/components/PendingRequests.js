import React, { Component } from 'react';
import { View, StyleSheet, Alert, ListView } from 'react-native';
import { ListItem, Icon } from 'react-native-elements';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import StarRating from 'react-native-star-rating';
import { updatePendingRequests, acceptPendingRequest } from '../actions/profile';
import StyledText from './StyledText';

import TopNavBar from './TopNavBar';

class PendingRequests extends Component {
  constructor(props) {
    super(props);
    this.state = { dataSource: new ListView.DataSource({ rowHasChanged: (r1, r2) => r1 !== r2 }).cloneWithRows(this.props.pendingRequests) };
  }

  componentDidMount() {
    this.props.updatePendingRequests(1);
  }

  componentWillReceiveProps(nextProps) {
    this.setState({ dataSource: this.state.dataSource.cloneWithRows(nextProps.pendingRequests) });
  }

  renderRow(u) {
    return (
      <ListItem
        key={u.user_id}
        title={u.first_name + ' ' + u.last_name}
        titleStyle={style.title}
        wrapperStyle={style.listItem}
        hideChevron
        subtitle={u.course_prefix + u.course_code}
        label={
          <View>
            <StarRating
              maxStars={5}
              rating={u.agg_tutor_rating / u.num_tutor_rating}
              starSize={10}
              starColor={'gold'}
              selectedStar={function() {}}
              disabled
            />
            <Icon
              name="reply"
              color="#42bcf4"
              size={30}
              containerStyle={style.icon}
              onPress={()=>{
                Alert.alert(
                  'Pending Request',
                  'Do you want to tutor ' + u.first_name + ' ' + u.last_name + '?',
                  [
                    { text: 'Yes', onPress: () => { this.props.acceptPendingRequest(u.id, u.user_id); } },
                    { text: 'No' }
                  ],
                  { cancelable: false }
                );
              }}
            />
          </View>
        }
      />
    );
  }

  render() {
    const { pendingRequests } = this.props;
    return (
      <View style={style.container}>
        <TopNavBar toggled={this.props.toggled} tabName={this.props.tabName} />
        <StyledText style={style.text}>You have {pendingRequests.length} requests waiting.</StyledText>
        <ListView
          dataSource={this.state.dataSource}
          renderRow={(row) => this.renderRow(row)}
        />
      </View>
    );
  }
}

const white = 'white';
const blue = '#8080ff';
const black = 'black';
const style = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: white,
  },
  text: {
    textAlign: 'center',
    marginTop:10,
    color: 'green',
    fontSize: 16
  },
  listItem: {
    marginRight: 20,
    marginLeft: 20,
    marginBottom:0
  },
  title:{ color: black, fontSize: 14 }
});

const mapStateToProps = (state) => {
  return { pendingRequests: state.profile.pendingRequests };
};

const mapDispatchToProps = (dispatch) => bindActionCreators({ updatePendingRequests, acceptPendingRequest }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(PendingRequests);
