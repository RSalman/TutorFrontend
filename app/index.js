import React, { Component } from 'react';
import { StyleSheet,TouchableHighlight, Text, View, Navigator } from 'react-native';

import HomeView from './views/HomeView';
import TutorsComponent from './components/TutorsComponent';
import LeilaFeatureView from './views/LeilaFeatureView';
import SalmanFeatureView from './views/SalmanFeatureView';
import MuraadFeatureView from './views/MuraadFeatureView';

export default class TutoringApp extends Component {
  render() {
    return (
  		<Navigator
        	initialRoute={{screen: 'HomeView'}}
        	renderScene={(route, nav) => {return this.renderScene(route, nav)}}  />
    );
  }

 renderScene(route,nav) {
    switch (route.screen) {
      	case "HomeView":
        	return <HomeView navigator={nav} />
      	case "TutorsComponent":
        	return <TutorsComponent navigator={nav} />
      	case "SalmanFeature":
        	return <SalmanFeatureView navigator={nav} /> 
        case "LeilaFeature":
        	return <LeilaFeatureView navigator={nav} /> 
        case "MuraadFeature":
        	return <MuraadFeatureView navigator={nav} /> 
      }
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  }
});
