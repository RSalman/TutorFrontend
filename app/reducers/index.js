import { combineReducers } from 'redux';
import tutors from './tutors';
import profile from './profile';

const rootReducer = combineReducers({
  tutors: tutors,
  profile: profile
});

export default rootReducer;
