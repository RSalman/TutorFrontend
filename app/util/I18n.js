import I18n from 'react-native-i18n';

import en from './locales/en';
import fr from './locales/fr';

I18n.fallbacks = true;

I18n.translations = {
  en,
  fr
};

global.I18n = I18n;
