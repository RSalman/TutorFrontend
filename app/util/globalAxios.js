import axios from 'axios';
import { Alert, NetInfo } from 'react-native';

axios.defaults.baseURL = 'http://cpserver.eastus.cloudapp.azure.com:2000/api/v1';
axios.defaults.headers.post['Content-Type'] = 'application/json';
axios.defaults.headers.get['Content-Type'] = 'application/json';
axios.defaults.headers.post.Accept = 'application/json';
axios.defaults.headers.get.Accept = 'application/json';

var connection = true;
function connectivityChange(isConnected) {
  connection = isConnected;
}

function showNoInternetDialog(config) {
  return new Promise(function(resolve, reject) {
    Alert.alert(
      I18n.t('globalWifiDialog.noInternetConnection'),
      I18n.t('globalWifiDialog.fixInternet'),
      [
        { text: I18n.t('retry'), onPress: () => resolve(axios(config)) },
      ],
      { cancelable: false }
    );
  });
}

NetInfo.isConnected.addEventListener('change', connectivityChange);

axios.interceptors.response.use(undefined, function(error) {
  return connection ? Promise.reject(error) : showNoInternetDialog(error.config);
});

global.axios = axios;
