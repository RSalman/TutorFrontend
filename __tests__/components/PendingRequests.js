import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import PendingRequests from '../../app/components/PendingRequests';
import '../../app/index';
import '../../app/util/globalAxios';
import '../../app/util/I18n';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

function setup(state) {
  const store = mockStore(state);
  const props = {
    updatePendingRequests: jest.fn(),
    store
  };
  return props;
}

describe('<PendingRequests />', () => {
  it('renders correctly', () => {
    const props = setup({
      session: {
        tutorMode: true,
      },
      profile: {
        profile: null,
        isLoading: false,
        requesting: false,
        requestSent: false,
        error: '',
        requestError: false,
        requestDeleted: false,
        pendingRequests: [],
        acceptedRequests: [],
        current_user: { id: null },
        access_token: null,
        courses_request_status: []
      }
    });
    const tree = renderer.create(
      <PendingRequests {...props} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
