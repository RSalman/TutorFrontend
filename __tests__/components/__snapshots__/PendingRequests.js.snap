exports[`<PendingRequests /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "flexDirection": "row",
        "marginTop": 20,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "fontFamily": "HelveticaNeue-Thin",
          },
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
            "fontSize": 20,
            "textAlign": "center",
          },
        ]
      } />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "fontFamily": "HelveticaNeue-Thin",
        },
        Object {
          "color": "green",
          "fontSize": 16,
          "marginTop": 10,
          "textAlign": "center",
        },
      ]
    }>
    You have 
    0
     request(s) waiting.
  </Text>
  <View>
    <RCTScrollView>
      <View />
    </RCTScrollView>
  </View>
</View>
`;
